test/unit/proxy_config_test.rb

```diff
@@ -113,6 +113,17 @@ def test_create_without_service_token
     assert proxy_config.valid?
   end
 
+  test 'missing api backend when service_mesh' do
+    service = FactoryBot.create(:simple_service)
+    proxy = FactoryBot.create(:proxy, service: service)
+    proxy.stubs(:service_mesh_integration?).returns(true)
+    proxy_config = FactoryBot.build(:proxy_config, environment: 'sandbox', proxy: proxy)
```

Comment:
I guess that the specific environment in this case does not matter.
```suggestion
    proxy_config = FactoryBot.build(:proxy_config, proxy: proxy)
```